packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1089 @@
+  overflowMode?: 'wrap' | 'truncate',
+  /** How selection should be displayed. */
+  selectionStyle?: 'checkbox' | 'highlight',
+  // TODO: will we contine with onAction or rename to onRowAction like it is in RAC?
+  /** Handler that is called when a user performs an action on a row. */
+  onAction?: (key: Key) => void,
+  /**
+   * Handler that is called when a user starts a column resize.
+   */
+  onResizeStart?: (widths: Map<Key, ColumnSize>) => void,
+  /**
+   * Handler that is called when a user performs a column resize.
+   * Can be used with the width property on columns to put the column widths into
+   * a controlled state.
+   */
+  onResize?: (widths: Map<Key, ColumnSize>) => void,
+  /**
+   * Handler that is called after a user performs a column resize.
+   * Can be used to store the widths of columns for another future session.
+   */
+  onResizeEnd?: (widths: Map<Key, ColumnSize>) => void,
+  /** The current loading state of the table. */
+  loadingState?: LoadingState,
+  /** Handler that is called when more items should be loaded, e.g. while scrolling near the bottom. */
+  onLoadMore?: () => any
+}
+
+// TODO: Note that loadMore and loadingState are now on the Table instead of on the TableBody
+export interface TableProps extends Omit<RACTableProps, 'style' | 'disabledBehavior' | 'className' | 'onRowAction' | 'selectionBehavior' | 'onScroll' | 'onCellAction' | 'dragAndDropHooks'>, StyleProps, S2TableProps {
+}
+
+let InternalTableContext = createContext<TableProps & {scale?: Scale, layout?: S2TableLayout<unknown>, setIsInResizeMode?:(val: boolean) => void, isInResizeMode?: boolean}>({});
+
+const tableWrapper = style({
+  width: 'full'
+});
+
+const table = style<TableRenderProps & S2TableProps>({
+  height: 'full',
+  userSelect: 'none',
+  minHeight: 0,
+  minWidth: 0,
+  fontFamily: 'sans',
+  fontWeight: 'normal',
+  overflow: 'auto',
+  backgroundColor: {
+    default: 'gray-25',
+    isQuiet: 'transparent',
+    forcedColors: 'Background'
+  },
+  outlineColor: {
+    default: 'gray-300',
+    isFocusVisible: 'focus-ring',
+    forcedColors: 'ButtonBorder'
+  },
+  outlineWidth: {
+    default: 1,
+    isQuiet: 0,
+    isFocusVisible: 2
+  },
+  outlineStyle: 'solid',
+  borderRadius: {
+    default: size(6),
+    isQuiet: 'none'
+  }
+}, getAllowedOverrides({height: true}));
+
+// component-height-100
+const DEFAULT_HEADER_HEIGHT = {
+  medium: 32,
+  large: 40
+};
+
+const ROW_HEIGHTS = {
+  compact: {
+    medium: 32, // table-row-height-medium-compact (aka component-height-100)
+    large: 40
+  },
+  regular: {
+    medium: 40, // table-row-height-medium-regular
+    large: 50
+  },
+  spacious: {
+    medium: 48, // table-row-height-medium-spacious
+    large: 60
+  }
+};
+
+type Scale = 'large' | 'medium';
+
+export class S2TableLayout<T> extends UNSTABLE_TableLayout<T> {
+  protected isStickyColumn(node: GridNode<T>): boolean {
+    if (node.props.isSticky) {
+      return true;
+    }
+
+    return false;
+  }
```

Comment:
```suggestion
  protected isStickyColumn(node: GridNode<T>): boolean {
    return node.props.isSticky
  }
```
Cast to `Boolean` if it can be undefined